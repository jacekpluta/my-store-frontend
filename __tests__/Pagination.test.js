import { mount } from "enzyme";
import Pagination, { PAGINATION_QUERY } from "../components/Pagination";
import { MockedProvider } from "@apollo/react-testing";

function makeMocksFor(length) {
  return [
    {
      request: { query: PAGINATION_QUERY },
      result: {
        data: {
          itemsConnection: {
            __typename: "aggregate",
            aggregate: {
              count: length,
              __typename: "count",
            },
          },
        },
      },
    },
  ];
}
let wrapper;
beforeEach(() => {
  wrapper = mount(
    <MockedProvider mocks={makeMocksFor(12)}>
      <Pagination page={1} />
    </MockedProvider>
  );
});

it("shows loading", async () => {
  expect(wrapper.text()).toContain("Loading...");
});

it("renders user nav component when signed in", async () => {
  await new Promise((resolve) => setTimeout(resolve, 0));
  wrapper.update();

  const pagination = wrapper.find('[data-test="pagination"]');
  expect(pagination).toMatchSnapshot();

  const allPages = wrapper.find(".allPages").text();
  expect(allPages).toEqual("3");
});

it("disables prev button on first page", async () => {
  const wrapper = mount(
    <MockedProvider mocks={makeMocksFor(12)}>
      <Pagination page={1} />
    </MockedProvider>
  );

  await new Promise((resolve) => setTimeout(resolve, 0));
  wrapper.update();

  const prev = wrapper.find("a.prev").prop("aria-disabled");
  const next = wrapper.find("a.next").prop("aria-disabled");
  expect(prev).toEqual(true);
  expect(next).toEqual(false);
});

it("disables next button on last page", async () => {
  const wrapper = mount(
    <MockedProvider mocks={makeMocksFor(12)}>
      <Pagination page={3} />
    </MockedProvider>
  );

  await new Promise((resolve) => setTimeout(resolve, 0));
  wrapper.update();

  const prev = wrapper.find("a.prev").prop("aria-disabled");
  const next = wrapper.find("a.next").prop("aria-disabled");
  expect(prev).toEqual(false);
  expect(next).toEqual(true);
});

it("enables all buttons if not on the first or list page", async () => {
  const wrapper = mount(
    <MockedProvider mocks={makeMocksFor(12)}>
      <Pagination page={2} />
    </MockedProvider>
  );

  await new Promise((resolve) => setTimeout(resolve, 0));
  wrapper.update();

  const prev = wrapper.find("a.prev").prop("aria-disabled");
  const next = wrapper.find("a.next").prop("aria-disabled");
  expect(prev).toEqual(false);
  expect(next).toEqual(false);
});
