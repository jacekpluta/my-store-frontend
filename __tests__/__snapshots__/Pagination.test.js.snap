// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user nav component when signed in 1`] = `
Array [
  <PaginationStyles
    data-test="pagination"
  >
    <div
      className="PaginationStyles-hhtwtv-0 BfNRm"
      data-test="pagination"
    >
      <Head>
        <_default
          headManager={Object {}}
          inAmpMode={false}
          reduceComponentsToState={[Function]}
        />
      </Head>
      <Link
        href={
          Object {
            "pathname": "/catalog",
            "query": Object {
              "page": 0,
            },
          }
        }
      >
        <a
          aria-disabled={true}
          className="prev"
          href="/items?page=0"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          ⬅ Prev
        </a>
      </Link>
      <p>
        Page 
        1
         of 
        <span
          className="allPages"
        >
          3
        </span>
      </p>
      <Link
        href={
          Object {
            "pathname": "/catalog",
            "query": Object {
              "page": 2,
            },
          }
        }
      >
        <a
          aria-disabled={false}
          className="next"
          href="/items?page=2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          ➡ Next
        </a>
      </Link>
    </div>
  </PaginationStyles>,
  <div
    className="PaginationStyles-hhtwtv-0 BfNRm"
    data-test="pagination"
  >
    <Head>
      <_default
        headManager={Object {}}
        inAmpMode={false}
        reduceComponentsToState={[Function]}
      />
    </Head>
    <Link
      href={
        Object {
          "pathname": "/catalog",
          "query": Object {
            "page": 0,
          },
        }
      }
    >
      <a
        aria-disabled={true}
        className="prev"
        href="/items?page=0"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        ⬅ Prev
      </a>
    </Link>
    <p>
      Page 
      1
       of 
      <span
        className="allPages"
      >
        3
      </span>
    </p>
    <Link
      href={
        Object {
          "pathname": "/catalog",
          "query": Object {
            "page": 2,
          },
        }
      }
    >
      <a
        aria-disabled={false}
        className="next"
        href="/items?page=2"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        ➡ Next
      </a>
    </Link>
  </div>,
]
`;
